### PR TITLE
docs(react-grid): update Plugin Overview guide

### PR DESCRIPTION
**Pull request by gsobolev**

### CODE REVIEW

File: packages/dx-react-grid/docs/guides/plugin-overview.md [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
 
-The plugin order is important. Plugins implementing an interface should be linked before the plugin that uses it. For example, a data processing plugin is based on some state, and should follow the appropriate state plugin. Some visualization plugins extend the `Table`'s functionality and should follow it in the code. This rule is demonstrated in the following example:
+Because of the piping principal the plugin order is very important. Plugins implementing an interface should be linked before the plugin that uses it. For example, a data processing plugin is based on some state, and should follow the appropriate state plugin. Some visualization plugins extend the `Table`'s functionality and should follow it in the code. This rule is demonstrated in the following example:
 
 .embedded-demo(grid-filtering/filter-row)
 
 NOTE: Refer to the plugin's reference for information on its dependencies.
 
 The data processing plugins' order is also important because they transform data in the same order they appear. For example, if the `IntegratedPaging` plugin precedes the `IntegratedFiltering`, the Grid filters the current page's data. Swap the plugins to paginate filtered data.
```
Comment (MaximKudriavtsev) [inline packages/dx-react-grid/docs/guides/plugin-overview.md]: I think that we should highlight this paragraph to `###The Data Processing Plugins Order` section and add to new paragraph `select-all-by-page` demo.

File: packages/dx-react-grid/docs/guides/plugin-overview.md [hunk 2]
```diff
@@ -1,6 +1,22 @@
 # React Grid - Plugin Overview
 
-The plugins that provide all the Grid's functionality can be divided into four groups:
+Plugins are components which should be placed inside Grid and responsible for enabling particular Grid features. The core principle used for supplying plugins with data is piping. This means that data flows consequently from the first plugin to the next one and so on. And this is where [plugin order](#pluginorder) comes to a scene.
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid/docs/guides/plugin-overview.md]: I think we can use this link [piping](https://en.wikipedia.org/wiki/Pipeline_(computing))

File: packages/dx-react-grid/docs/guides/plugin-overview.md [hunk 3]
```diff
@@ -11,19 +27,9 @@ Note that the plugins are composable and can be nested into each other.
...
-.embedded-demo(grid-filtering/filter-row)
-
-NOTE: Refer to the plugin's reference for information on its dependencies.
-
-The data processing plugins' order is also important because they transform data in the same order they appear. For example, if the `IntegratedPaging` plugin precedes the `IntegratedFiltering`, the Grid filters the current page's data. Swap the plugins to paginate filtered data.
-
 ## UI Plugins
 
-The Grid's UI plugins use templates to render the UI. A template is a function that returns a [ReactElement](https://facebook.github.io/react/docs/react-api.html#createelement) depending on the argument values. You can implement your templates or use one of the predefined template suites:
+The Grid's UI plugins use special components to render the UI. You can implement your own suite of components or chose between the predefined ones:
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid/docs/guides/plugin-overview.md]: > You can implement your own suite of components or chose between the predefined ones

I'd change the priority... something like this:

You can use one of the predefined component suites or implement your own one.

Comment (gsobolev) [reply packages/dx-react-grid/docs/guides/plugin-overview.md]: Did not think about that, simply updated the existing sentence. @viterobk, what do you think at this point?

Comment (viterobk) [reply packages/dx-react-grid/docs/guides/plugin-overview.md]: Currently, the sentence brings a reader to the components list that follows the sentence. Transposition of the "choose..." and "implement..." parts, will require the additional connection between the sentence and the list, like "The following predefined components are available". For me, it's better to leave this paragraph as is.

**Pull request merged**